app/lib/connection.ts

```diff
@@ -1,20 +1,33 @@
 import { BasicCredentialHandler } from "azure-devops-node-api/handlers/basiccreds";
 
-import url = require("url");
 import apim = require("azure-devops-node-api/WebApi");
 import apibasem = require("azure-devops-node-api/interfaces/common/VsoBaseInterfaces");
 import trace = require("./trace");
 
 export class TfsConnection {
-	private parsedUrl: url.Url;
+	private parsedUrl: URL;
 
 	private accountUrl: string;
 	private collectionUrl: string;
 
 	constructor(private serviceUrl: string) {
-		this.parsedUrl = url.parse(this.serviceUrl);
+		// Parse URL, but handle failures gracefully to mimic url.parse() behavior
+		try {
+			this.parsedUrl = new URL(this.serviceUrl);
+		} catch (error) {
+			// Mimic url.parse() behavior for invalid URLs
+			// url.parse() would return an object with null/empty values instead of throwing
+			this.parsedUrl = {
+				protocol: this.serviceUrl && this.serviceUrl.includes('://') ? this.serviceUrl.split('://')[0] + ':' : '',
+				host: null,
+				hostname: null,
+				pathname: this.serviceUrl && !this.serviceUrl.includes('://') ? this.serviceUrl : '',
+				search: '',
+				hash: ''
+			} as any;
+		}
```

Comment:
This method already has error handling and throws. If a user submits an invalid URL like `/tfs/orgName` we can just throw earlier, with a humanized version of that message. No need to limp it through only to fail it later.

Throwing earlier also fixes a bug on line 35 because of a poorly-ordered if clause.

URL input: `/tfs/orgName`

31: `this.accountUrl = this.parsedUrl.protocol + "//" + this.parsedUrl.host;` => `null//null`
35: `this.accountUrl += "/" + "tfs";`
=> `null//null/tfs`